app/components/post_list/post/body/content/message_attachments/message_attachment.test.tsx

```diff
@@ -0,0 +1,324 @@
+        mockAttachmentFooter(props);
+        return null;
+    };
+});
+
+const mockAttachmentActions = jest.fn().mockReturnValue(null);
+jest.mock('./attachment_actions', () => {
+    return function MockAttachmentActions(props: {actions: PostAction[]; postId: string; theme: Theme}) {
+        mockAttachmentActions(props);
+        return null;
+    };
+});
+
+jest.mock('./attachment_author', () => {
+    return function MockAttachmentAuthor() {
+        return null;
+    };
+});
+
+jest.mock('./attachment_image', () => {
+    return function MockAttachmentImage() {
+        return null;
+    };
+});
+
+jest.mock('./attachment_pretext', () => {
+    return function MockAttachmentPretext() {
+        return null;
+    };
+});
+
+jest.mock('./attachment_text', () => {
+    return function MockAttachmentText() {
+        return null;
+    };
+});
+
+jest.mock('./attachment_title', () => {
+    return function MockAttachmentTitle() {
+        return null;
+    };
+});
+
+// Mock the URL validation function
+jest.mock('@utils/url', () => ({
+    isValidUrl: jest.fn((url) => url === 'https://example.com/image.png'),
+}));
+
+import MessageAttachment from './message_attachment';
+
+describe('MessageAttachment', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    const baseProps = {
+        attachment: {
+            id: 1,
+            text: 'This is the text of an attachment',
+            thumb_url: 'https://example.com/image.png',
+            author_name: 'Author Name',
+            author_icon: 'https://example.com/author.png',
+            title: 'Attachment Title',
+            title_link: 'https://example.com',
+            fields: [
+                {
+                    title: 'Field Title',
+                    value: 'Field Value',
+                    short: true,
+                },
+            ],
+            actions: [
+                {
+                    id: 'action1',
+                    name: 'Action 1',
+                    type: 'button',
+                    integration: {
+                        url: 'https://example.com/action',
+                    },
+                },
+            ],
+            footer: 'Attachment Footer',
+            footer_icon: 'https://example.com/footer.png',
+            image_url: 'https://example.com/image.jpg',
+        },
+        channelId: 'channel-id',
+        postId: 'post-id',
+        location: 'CENTER',
+        theme: Preferences.THEMES.denim,
+        metadata: {
+            images: {
+                'https://example.com/image.jpg': {
+                    width: 200,
+                    height: 200,
+                    format: 'jpg',
+                    frame_count: 0,
+                },
+            },
+        },
+    };
+
+    test('it renders without crashing with full attachment data', () => {
+        const wrapper = render(<MessageAttachment {...baseProps}/>);
+        expect(wrapper.toJSON()).toBeTruthy();
```

Comment:
(0/5) we should start using screen vs wrapper.

https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen

> The benefit of using screen is you no longer need to keep the render call destructure up-to-date as you add/remove the queries you need. You only need to type screen. and let your editor's magic autocomplete take care of the rest.

Reply:
> Without the fix, would this fail?

It shouldn't, as all fields have value, but if something crashes, this should pick it up, which could be useful in other scenarios